reader-osm/src/test/java/com/graphhopper/GraphHopperIT.java

```diff
@@ -1381,4 +1381,32 @@ private GHResponse calcCurbsidePath(GraphHopper tmpHopper, GHPoint source, GHPoi
         return tmpHopper.route(req);
     }
 
+    @Test
+    public void testCHWithFiniteUTurnCostsAndMissingWeighting() {
+        GraphHopper h = new GraphHopperOSM().
+                setDataReaderFile(DIR + "/monaco.osm.gz").
+                setCHEnabled(true).
+                setGraphHopperLocation(tmpGraphFile).
+                setEncodingManager(EncodingManager.create("car|turn_costs=true"));
+        h.getCHFactoryDecorator()
+                .setCHProfileStrings("fastest|u_turn_costs=40")
+                .setEdgeBasedCHMode(CHAlgoFactoryDecorator.EdgeBasedCHMode.EDGE_OR_NODE);
+        h.importOrLoad();
+
+        GHPoint p = new GHPoint(43.73397, 7.414173);
+        GHPoint q = new GHPoint(43.73222, 7.415557);
+        GHRequest req = new GHRequest(p, q);
+        // note that we do *not* set the weighting on the request, it will be determined automatically from the
+        // CH profile, see #1788
+        req.setWeighting("fastest");
```

Comment:
I do not understand this comment as req.setWeighting is called. Can you explain :) ?

Reply:
Oh thanks for the review! Yes this is a mistake, I probably added this line there and back while testing, will remove it.